tests/test_dot.py

```diff
@@ -0,0 +1,114 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import os
+import tempfile
+import unittest
+
+import retworkx
+
+
+class TestDot(unittest.TestCase):
+    def setUp(self):
+        fd, self.path = tempfile.mkstemp()
+        os.close(fd)
+        os.remove(self.path)
+
+    def test_graph_to_dot(self):
+        graph = retworkx.PyGraph()
+        graph.add_node({'color': 'black', 'fillcolor': 'green',
+                        'label': "a", 'style': 'filled'})
+        graph.add_node({'color': 'black', 'fillcolor': 'red',
+                        'label': "a", 'style': 'filled'})
+        graph.add_edge(0, 1, dict(label='1', name='1'))
+        expected = (
+            "graph {\n0 [color=black, fillcolor=green, label=a, style=filled"
+            "];\n1 [color=black, fillcolor=red, label=a, style=filled];"
+            "\n0 -- 1 [label=1, name=1];\n}\n")
+        res = graph.to_dot(lambda node: node, lambda edge: edge).decode('utf8')
+        self.assertEqual(expected, res)
+
+    def test_digraph_to_dot(self):
+        graph = retworkx.PyDiGraph()
+        graph.add_node({'color': 'black', 'fillcolor': 'green',
```

Comment:
How much would users normally interact with these dot options?

Reply:
It depends on how much they want to format the output. If you just did `to_dot(lambda _: {}, lambda _: {})` it works fine but there wouldn't be any formatting just simple lines and no annotation except for the node ids. Something like:

![test_dot](https://user-images.githubusercontent.com/2447371/88587616-34b68b80-d024-11ea-9d85-5ed739c63c6d.png)

Are you thinking that we should make the `*attrs` parameters default to None and make formatting opt-in?

Reply:
@lcapelluto I modified the methods so that `node_attr` and `edge_attr` are optional and default to None in https://github.com/Qiskit/retworkx/pull/103/commits/c4536b863577412e7bf05c0abda24de01e0ff74a so now you can just call `PyGraph.to_dot()` to generate a dot file without any formatting.

Reply:
Cool! I like that a lot. Yeah, it felt like it might be nice to be able to separate building a graph and formatting a graph. I'd usually build one, do some operations, and maybe later want to display it. But I see now that it's just because the function passed is `lambda node: node`, and the user could do something different, like you showed in the other comment thread. Neat!